### DISCUSSION TITLE
Notification is sent to all users instead of selected collection of notifiables

### DISCUSSION THREAD

**Discussion by baszach** [laravel/nova-issues#5094] [Q&A]
- Laravel Version: 9.17
- Nova Version: 4.12.6
- PHP Version: 8.1.6

Description:
I use the following code snippet to send notifications to my **admin users only**. I have tested the method `User::adminUsers()` and it returns only my admins correctly.

Still the notification is being sent to all notifiable users in my application. Is this some sort of bug or am I using the notification wrongly?

```
use Illuminate\Support\Facades\Notification;
use Laravel\Nova\Notifications\NovaNotification;

.....

Notification::send(
       User::adminUsers(),
       NovaNotification::make()
           ->message('myMessage')
           ->url('myUrl')
           ->type('info')
   );
```

**Comment by crynobone** [comment]
Unable to reproduce the issue, please provide full reproducing repository based on fresh installation as suggested in the bug report template (or you can refer to https://github.com/nova-issues for example)

**Comment by baszach** [comment]
An important detail that I just discovered while debugging:
I have two different user models. The one is `User` and the other one `Contractor`. The `notifiable_type` column in the `nova_notifications` table has the value `users` and the `notifiable_id` is `1`. If I log in with the `Contractor` user with the ID 1, I can still see the notification although he shouldn't be of `notifiable_type` `users`. He should have the `notifiable_type` `contractors` and thus not see this notification..

**Comment by crynobone** [comment]
More reason a reproducing code is needed to understand what's is wrong. At the moment, I can only make assumption that this isn't a bug but a usage error
